source/Octopus.Client/OctopusAsyncClient.cs

```diff
@@ -686,6 +724,50 @@ private async Task<T> ReadResponse<T>(HttpResponseMessage response)
             }
         }
 
+        private OctopusClientOptions CreateClientOptions(SpaceContext spaceContext)
+        {
+            return new OctopusClientOptions()
+            {
+                Proxy = clientOptions.Proxy,
+                ProxyPassword = clientOptions.ProxyPassword,
+                ProxyUsername = clientOptions.ProxyUsername,
+                Timeout = clientOptions.Timeout,
+                SpaceContext = spaceContext,
+                CookieContainer = this.cookieContainer
+            };
+        }
+
+        private void ValidateSpaceId(string spaceId)
+        {
+            if (string.IsNullOrEmpty(spaceId))
+            {
+                throw new ArgumentException("spaceId cannot be null");
+            }
+
+            if (spaceId == "all")
```

Comment:
hmmm, maybe at least put `"all"` in a global const string somewhere, cause it's also used on line 755?